doc/development/plugins.rst

```diff
@@ -195,6 +204,164 @@ or can include in-built transforms such as:
 * :func:`pennylane.devices.preprocess.validate_adjoint_trainable_params`
 * :func:`pennylane.devices.preprocess.no_sampling`
 
+PennyLane also provides a default implementation of :meth:`~pennylane.devices.Device.preprocess`
+which should be sufficient for most plugin devices. To take advantage of this default implementation,
+you need to define a configuration file for your device as described below.
+
+.. _device_capabilities:
+
+Device Capabilities
+-------------------
+
+Optionally, you can add a ``config_filepath`` class variable pointing to your configuration file. This file
+should be a `toml file <https://toml.io/en/>`_ that describes which gates and features are supported
+by your device, i.e., what the :meth:`~pennylane.devices.Device.execute` method accepts.
+
+.. code-block:: python
+
+    from os import path
+    from pennylane.devices import Device
+
+    class MyDevice(Device):
+        """My Documentation."""
+
+        config_filepath = path.join(path.dirname(__file__), "relative/path/to/config.toml")
+
+This configuration file will be loaded into another class variable :attr:`~pennylane.devices.Device.capabilities`
```

Comment:
Lets add a mention that the file should be imported as package data in `setup.py` and show how to do so.

Reply:
Done